src/machines/appMachine.ts

```diff
@@ -0,0 +1,30 @@
+import { ActorRefFrom, createActor, setup } from 'xstate'
+import { authMachine } from './authMachine'
+import { useSelector } from '@xstate/react'
+import { ACTOR_IDS } from './machineConstants'
+
+const appMachine = setup({
+  actors: {
+    [ACTOR_IDS.AUTH]: authMachine,
+  },
+}).createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5gF8A0IB2B7CdGgAoBbAQwGMALASwzAEp8QAHLWKgFyqw0YA9EAjACZ0AT0FDkU5EA */
+  id: 'modeling-app',
+  invoke: [
+    {
+      src: ACTOR_IDS.AUTH,
+      systemId: ACTOR_IDS.AUTH,
+    },
+  ],
+})
+
+export const appActor = createActor(appMachine).start()
+
+export const authActor = appActor.system.get(ACTOR_IDS.AUTH) as ActorRefFrom<
+  typeof authMachine
+>
```

Comment:
TypeScript can't infer the type properly I take it?

Reply:
No, this is my biggest beef so far with trying to make use of the XState actor system approach. Maybe I'm missing something but `appActor.system.get()` has this type signature according to my intellisense:
`(property) ActorSystem<any>.get: <string>(key: string) => any`

Reply:
Yeah we'll look together at a better solution - not worth blocking IMO.